
library(ggplot2)
library(reshape2)

allsponge_df=melt(allsponge)
View(allsponge_df)
names(allsponge_df)=c('keywords','sample','frequency')
write.csv(allsponge_df, file="revisedfunctions.swissprotkeywords.top50.reordered.csv")
#####add the species
allspongereloaded=read.csv("revisedfunctions.swissprotkeywords.top50.reordered.csv")
allspongereloaded$sample<-factor(allspongereloaded$sample, levels=c("GBR_CALLY_37","GBR_CALLY_46","GBR_CALLY_66","GBR_CAR_14","GBR_CAR_2","GBR_CAR_24","GBR_CINA_35","GBR_CINA_60","GBR_CINA_61","GBR_CYM_25","GBR_CYM_30","GBR_CYM_9","GBR_ECHI_13","GBR_ECHI_34","GBR_ECHI_42","GBR_IAN_31","GBR_IAN_38","GBR_IAN_52","GBR_LAM_44","GBR_LAM_48","GBR_LAM_71","GBR_PIP_19","GBR_PIP_58","GBR_PIP_63","GBR_STY_33","GBR_STY_70","RS_AOC_150","RS_AOC_153","RS_AOC_202","RS_CAR_111","RS_CAR_112","RS_CAR_129","RS_CRE_178","RS_CRE_184","RS_CRE_214","RS_HYR_175","RS_HYR_180","RS_MYC_127","RS_MYC_176","RS_MYC_197","RS_NIP_200","RS_NIP_211","RS_XES_102","RS_XES_136","GBR_SW","RS_SW"))
allspongereloaded$keywords<-factor(allspongereloaded$keywords, levels=rev(c("Viral receptor tropism switching","Viral long flexible tail ejection system","Viral short tail ejection system","Degradation of host lipopolysaccharides during virus entry","Viral genome ejection through host cell envelope","Degradation of host chromosome by virus","Bacterial host gene expression shutoff by virus","Viral contractile tail ejection system","Virus exit from host cell","Viral attachment to host entry receptor","Host cell lysis by virus","Degradation of host cell envelope components during virus entry","Restriction-modification system evasion by virus","Degradation of host peptidoglycans during virus entry",
                                                                            "Viral DNA replication","Viral genome excision","Hydroxylation","Evasion of bacteria-mediated translation shutoff by virus","Transposition","Late protein","Primosome","DNA-directed DNA polymerase","Covalent protein-DNA linkage","Bacteriolytic enzyme","Transposable element","DNA invertase",
                                                                            "Collagen","Viral tail tube protein","Viral tail assembly","Viral genome packaging","Viral capsid assembly","T=1 icosahedral capsid protein","Viral tail protein","Basement membrane","Viral baseplate protein","Viral tail fiber protein","Extracellular matrix","T=7 icosahedral capsid protein",
                                                                            "Restriction system","Host thylakoid","Chromate resistance","SOS mutagenesis","Cobalamin","Cadmium resistance","Nylon degradation","TonB box","PQQ","Limb-girdle muscular dystrophy","Ehlers-Danlos syndrome","Cardiomyopathy")))
allspongereloaded$species<-factor(allspongereloaded$species, levels=c("Callyspongia sp.","C. foliascens","C. schulzei","C. marshae","E. isaaci","I. basta","L. herbacea","P. candelabra","S. carteri","A. ochracea","Carteriospongia sp.","C. cyathophora","H. erectus","Mycale sp.","N. rowi","X. testudinaria","GBR Seawater","Red Sea Seawater"))
#                                                                                                                                  Scale for 'colour' is already present. Adding another scale for 'colour', which will replace the existing scale.
#p <- ggplot(isme2016_df, aes(y=keywords,x=sample,size=frequency))+geom_point()
#p <- p+ scale_color_manual()
#ggplot(isme2016_df, aes(x=keywords,y=sample,size=frequency))+geom_point()+scale_size_area(max_size=10,trans="log10")+theme(axis.title.x=element_blank(), axis.ticks.y=element_blank(), text=element_text(size=20), axis.text.x=element_blank(), axis.ticks.x=element_blank(), panel.background=element_blank(), axis.title.y=element_blank())+ coord_fixed(ratio = 1.2))
#+ggplot(isme2016species, aes(x=keywords,y=sample,size=frequency, color=species))+geom_point()+scale_size_area(max_size=10,trans="log10")+theme(axis.title.x=element_blank(), axis.ticks.y=element_blank(), text=element_text(size=20), axis.text.x=element_blank(), axis.ticks.x=element_blank(), panel.background=element_blank(), axis.title.y=element_blank())+ coord_fixed(ratio = 1.2)
#aes(x=keywords,y=sample,size=frequency, color=keywords)  scale_size_area(max_size=8)
allspongereloaded <-subset(allspongereloaded,frequency > 2) 
ggplot(allspongereloaded, aes(y=keywords,x=sample,size=frequency,color=species))+geom_point()+scale_size_area(max_size=5)+theme(axis.title.x=element_blank(), axis.ticks.y=element_blank(), text=element_text(size=8), axis.text.x=element_blank(), axis.ticks.x=element_blank(), panel.background=element_blank(), axis.title.y=element_blank())+ coord_fixed(ratio = 0.8)+scale_color_manual(values=c("steelblue3","skyblue","aquamarine","lightcyan3","navyblue","slateblue1","azure3","purple4","darkmagenta","plum3","darkgreen","grey40","grey70","darkseagreen","forestgreen","olivedrab","royalblue","royalblue3"))
#ggplot(isme2016species, aes(y=keywords,x=sample,size=frequency,color=species))+geom_point()+scale_size_area(max_size=5)+theme(axis.title.x=element_blank(), axis.ticks.y=element_blank(), text=element_text(size=11), axis.text.x=element_blank(), axis.ticks.x=element_blank(), panel.background=element_blank(), axis.title.y=element_blank())+ coord_fixed(ratio = 0.7)+scale_color_manual(values=c("magenta3","plum2","mediumpurple","violetred1","forestgreen","turquoise","royalblue","royalblue3"))
